### DISCUSSION TITLE
Unable to make page transitions using useLocation() in place of __RouterContext from 'react-router-dom'

### DISCUSSION THREAD

**Discussion by lfranckx** [pmndrs/react-spring#1892] [Support]
I am trying to create animated page transitions using useTransition(). I am using useLocation() provided from react-router-dom in order to get the pathname from the location object. This error constantly comes up I am assuming because the location object is null until the component is rendered. `TypeError: Cannot read properties of undefined (reading 'pathname')`

Sorry just to clarify my useLocation() is within the Router. I have defined Router inside of the index.js file where it is rendering the App component

```
import { useTransition, animated } from 'react-spring';
import { withRouter, Switch, Route, useLocation } from 'react-router-dom';

function App() {
  const location = useLocation();
  const transitions = useTransition(location, (location) => location.pathname, {
    from: { opacity: 0 },
    enter: { opacity: 1 },
    leave: { opacity: 0, },
  });

  return (
    {transitions.map(({ item, props, key }) => (
      <animated.div key={key} style={props}>
        <Switch location={item}>
          <Route path='/' component={Home} exact />
          <Route path='/about' component={About} />
        </Switch>
      </animated.div> ))}
  );
}

export default withRouter(App);
```
<img width="796" alt="Screen Shot 2022-05-11 at 9 23 01 PM" src="https://user-images.githubusercontent.com/52330544/167979433-a0c48d9d-5c78-4438-aa7d-4898cea71108.png">

**Comment by joshuaellis** [comment]
What version of react-spring are you using?

**Reply by joshuaellis** [reply]
Based on you using the `v9` of the library see this code snippet @lfranckx 

```jsx
import { useTransition, animated } from 'react-spring';
import { withRouter, Switch, Route, useLocation } from 'react-router-dom';

function App() {
  const location = useLocation();
  const transitions = useTransition(location, {
    keys: (location) => location.pathname,
    from: { opacity: 0 },
    enter: { opacity: 1 },
    leave: { opacity: 0, },
  });

  return (
    <>
    {transitions((styles, item) => (
      <animated.div style={styles}>
        <Switch location={item}>
          <Route path='/' component={Home} exact />
          <Route path='/about' component={About} />
        </Switch>
      </animated.div> ))}
    </>
  );
}

export default withRouter(App);
```

**Reply by lfranckx** [reply]
@joshuaellis It's not able to map transitions. 

<img width="688" alt="Screen Shot 2022-05-13 at 9 30 42 AM" src="https://user-images.githubusercontent.com/52330544/168305818-8fb83d42-ae15-46cd-bf5d-4533cb8233b9.png">

**Reply by joshuaellis** [reply]
I've edited it.

**Comment by lfranckx** [comment]
"react-spring": "^9.4.5"
